src/scenario/data.h

```diff
@@ -81,10 +81,35 @@ enum {
     ALLOWED_BUILDING_MONUMENTS = 48,
 };
 
-struct win_criteria_t {
+#define MAX_WIN_CRITERIA 10
+
+typedef enum {
+    WIN_CRITERIA_NONE,
+    WIN_CRITERIA_POPULATION_MINIMUM,
+    WIN_CRITERIA_CULTURE,
+    WIN_CRITERIA_PROSPERITY,
+    WIN_CRITERIA_PEACE,
+    WIN_CRITERIA_FAVOR,
+    WIN_CRITERIA_SURVIVAL_YEARS,
+    WIN_CRITERIA_TIME_LIMIT,
+
+    // TODO: Try to implement these
+    WIN_CRITERIA_BUILD_X_BUILDINGS, // Build at least a quantity of a specific building type
+    WIN_CRITERIA_EVOLVE_X_HOUSES,   // Build at least X houses of the given level or higher
+    WIN_CRITERIA_EARN_X_DENARII,    // Have at least X denarii in city coffers
+    WIN_CRITERIA_SELL_X_GOODS,      // Sell at least X of the specific trade good to any trade partner
```

Comment:
Yearly or total?
Perhaps WIN_CRITERIA_SELL_X_GOODS_IN_YEAR for selling more than X of the target good in a single year (to all trade partners) ?
Perhaps in future things like WIN_CRITERIA_SELL_X_GOODS_SUSTAINED can be selling more than X each year for Y years straight.

Reply:
I was thinking along the lines of "Sell 500 weapons", which would need to be accomplished over several years. My big thought here is that eventually trade routes could become fully dynamic: If you sell many items, the capacity of the route will go up (and the price of the good will go down). If you consistently under-sell, maybe the price will go up to entice you to start selling. Same with imports, where the capacity can increase (and price can increase) if you keep maxing out your buys. That way if a .map file has trade events only for 10 years, the system can continue to change over time if you go longer than 10 years. Also with repeated emporer requests, etc. If you sell a lot of Oil, the emporer will ask for a freebie, etc.